lib/sendf.c

```diff
@@ -360,7 +364,7 @@ ssize_t Curl_send_plain(struct connectdata *conn, int num,
      available. */
   pre_receive_plain(conn, num);
 
-#ifdef MSG_FASTOPEN /* Linux */
+#if defined(MSG_FASTOPEN) && !defined(TCP_FASTOPEN_CONNECT) /* Linux */
   if(conn->bits.tcp_fastopen) {
     bytes_written = sendto(sockfd, mem, len, MSG_FASTOPEN,
                            conn->ip_addr->ai_addr, conn->ip_addr->ai_addrlen);
```

Comment:
although not specifically related to your changes i'd like to address how depending on which fastopen is used tcp_fastopen may be set false. For example `conn->bits.tcp_fastopen = FALSE;` in the line below this line (I can't select it in the reviewer because it's too out of context). Yet Mac TFO and now this new TFO it says true. What concerns me is other code seem to expect it one way or the other such as 
https://github.com/curl/curl/blob/curl-7_56_1/lib/connect.c#L675
https://github.com/curl/curl/blob/curl-7_56_1/lib/connect.c#L780
was the intention for tcp_fastopen to always be set false or only set false in certain circumstances?

Reply:
In this case tcp_fastopen is set to false because the sendto() logic only needs to be run once per connection. In the other cases (macOS and TCP_FASTOPEN_CONNECT) there is no sendto() logic.

We could set it to false for the other cases immediately after connect(), but I don't think it's necessary.